### PR TITLE
fix: recover from poison message [TP-84907]

### PR DESCRIPTION
**Pull request by cboulanger23**
- [X] 🐛 Bug fix
- [ ] 🎁 New feature
- [ ] 💎 Refactor
- [ ] 🧹 Chore

### Related Defect(s), Issue(s) or Task(s)
https://pas.tpondemand.com/entity/84907-file-download-on-ario-is-very

### Why?
When an exception occurs in a rabbitMQ consumer handler function, the message is left unAck which causes the broker to close the connection after consumer_timeout expires (2h in prod).  The client side is unaware of the closure and keeps using the limbo connection for consuming which results in nothing being received.

### How?
Trap uncaught exceptions in consumer handlers and NACK the message up to 3 times.  After that, reject the message so that it is no longer being send to consumers.

- [ ] :godmode: None
- [ ] :suspect: Low
- [ ] 🤨 Medium  [Only is used in case of existing error!  So, should cause more harm that the actual issue]
- [ ] :rage3: High

### Checklist:
- [ ] I have **added tests** to cover my changes.
- [ ] I have **adjusted e2e** to cover my changes.

### @Mentions:
@kirksmithzt, @raj-pas 

### Screenshots:
- [X] No screenshots are required

### CODE REVIEW

File: lib/hopper.rb [hunk 1]
```diff
@@ -257,9 +294,12 @@ def create_channel
 
     def thread_local_variable(var_name, &block)
       result = Thread.current[var_name]
+      thread_id = Thread.current.object_id
```
Comment (cboulanger23) [inline lib/hopper.rb:297]: These are to debug the other issue we have in production with producers.

**Review by TC-SJ-Yoon — APPROVED** [review]

**Review by LauraMtzG — APPROVED** [review]

**Review by raj-pas — APPROVED** [review]

**Review by kirksmithzt — APPROVED** [review]
one question inline (for my own understanding)

File: spec/hopper_spec.rb [hunk 2]
```diff
@@ -340,6 +340,7 @@ def handle_object_created(_event_type, _event, _source); end
       expect(described_class.queue.message_count).to be_zero
     end
 
+    # rubocop:disable RSpec/ExampleLength
```
Comment (kirksmithzt) [inline spec/hopper_spec.rb:343]: that is a dumb rule - we could configure to ignore that one

**Review by kirksmithzt — APPROVED** [review]

File: spec/hopper_spec.rb (hunk 2 above)
Comment (cboulanger23) [reply spec/hopper_spec.rb:342]: I agree !

Comment (cboulanger23) [reply spec/hopper_spec.rb]: done

**Pull request merged**